fe/fe-core/src/main/java/org/apache/doris/planner/RuntimeFilterGenerator.java

```diff
@@ -122,6 +122,24 @@ private RuntimeFilterGenerator(Analyzer analyzer) {
         bloomFilterSizeLimits = new FilterSizeLimits(sessionVariable);
     }
 
+    static class HasConjunctPredicate implements com.google.common.base.Predicate<PlanNode> {
+        @Override
+        public boolean apply(PlanNode arg) {
+            // for simplicity, skip join node( which contains more than 1 tuple id )
+            return arg.getTupleIds().size() == 1 && !arg.conjuncts.isEmpty();
+        }
+    }
+
+    private static HashSet<TupleId> collectAllTupleIdsHavingConjunct(PlanNode rootNode) {
+        HashSet<TupleId> tupleIds = new HashSet<>();
+        List<PlanNode> nodes = Lists.newArrayList();
+        rootNode.collectAll(new HasConjunctPredicate(), nodes);
```

Comment:
Replace `nodes` with `RuntimeFilterGenerator::tupleHasConjuncts` , which condenses `collectAllTupleIdsHavingConjunct` into one line of code and puts it into `generateRuntimeFilters`.

Reply:
Done